File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cs
using System;
using System.Collections.Generic;
public class Solution {
    public bool solution(string s) {
  
        //스택 활용
        Stack<bool> temp = new Stack<bool>();
         foreach (char c in s){
            if(c=='(')
                temp.Push(true);
            else
           {  
             if(temp.Count == 0) return false;
             temp.Pop();
           }
             
         }
        return temp.Count == 0? true:false;
        //알고리즘
        /*int count;
        count=0;
        foreach (char c in s){
            if(count<0) return false;
            
            if(c=='(')
                count++;
            else
                 count--;
        }
        if (count==0)
        return true;
        else
             return false;*/
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;

      int count;
        count=0;
        for (char c : s){
            if(count<0) return false;
            
            if(c=='(')
                count++;
            else
                 count--;
        }
        if (count==0)
            return true;
  
        return false;

    return answer;
}